Dockerfile

```diff
@@ -0,0 +1,36 @@
+# Build stage
+FROM python:3.10-slim AS build
+
+# Set work directory
+WORKDIR /app
+
+# Copy and install dependencies
+COPY requirements.txt .
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Copy application files
+COPY . .
```

Comment:
Copying the entire application directory in the build stage is unnecessary if you are only installing dependencies. Consider copying only the requirements.txt file in the build stage.